repl/src/main/scala/com/cloudera/livy/repl/SparkInterpreter.scala

```diff
@@ -84,12 +88,33 @@ class SparkInterpreter(conf: SparkConf) extends Interpreter {
       setContextClassLoaderMethod.invoke(sparkIMain)
 
       sparkContext = SparkContext.getOrCreate(conf)
-
+      if (conf.getBoolean("spark.repl.enableHiveContext", false)) {
```

Comment:
You can replace this with a check that `hive-site.xml` is in the application's classpath, which is basically the only way this will work anyway. (See my earlier comment about distributing `hive-site.xml`.)